### PR TITLE
Lesson23

### PR DESCRIPTION
**Pull request by VasilyevDmitry**

### CODE REVIEW

File: src/pageObjects/formSignUp/components/SignUpPopup.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+     this.fieldLastName = page.locator('#signupLastName')
+     this.fieldEmail = page.locator('#signupEmail')
+     this.fieldPassword = page.locator('#signupPassword')
+     this.fieldRepeatPassword = page.locator('#signupRepeatPassword')
+     this.btnRegister = page.getByRole('button', { name: 'Register'})
+     this.btnMyProfileIsVisible = page.getByRole('button', { name: 'My profile'})
+     this.textInvalidFeedback = page.locator('.invalid-feedback')
+     }
+ 
+     async fill_Name (name) {
```
Comment (sttaran) [inline src/pageObjects/formSignUp/components/SignUpPopup.js]: use camelCase for variables and methods name

File: src/pageObjects/formSignUp/components/SignUpPopup.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+     async fill_LastName (lastName) {
+          lastName && await this.fieldLastName.fill(lastName)
+     }
+
+     async fill_Email (email) {
+          email && await this.fieldEmail.fill(email)
+     }
+
+     async fill_Password (password) {
+          password && await this.fieldPassword.fill(password)
```
Comment (sttaran) [inline src/pageObjects/formSignUp/components/SignUpPopup.js]: this approach makes sense when you accept more the one argument

remove `password &&`

Comment (sttaran) [reply src/pageObjects/formSignUp/components/SignUpPopup.js]: same for other methods

File: src/pageObjects/formSignUp/FormSignUp.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import BasePage from "../BasePage.js";
+
+export default class FormSignUp extends BasePage {
```
Comment (sttaran) [inline src/pageObjects/formSignUp/FormSignUp.js]: FormSignUp sounds like a component name

Comment (sttaran) [reply src/pageObjects/formSignUp/FormSignUp.js]: rename it sot sounds like a page name

File: src/pageObjects/BasePage.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import {expect} from "@playwright/test";
+import SignUp from "../components/SignUP.js";
+
+export default class BasePage {
+    constructor(page, url, waitPageLocator) {
+        this._page = page
+        this._url = url
+        this._waitPageLocator = waitPageLocator
+        this.signUp = new SignUp(page)
```
Comment (sttaran) [inline src/pageObjects/BasePage.js]: this.signUp = new SignUp(page)

it should not be at BasePage because this popup appears only on welcome page

**Review by sttaran — CHANGES_REQUESTED** [review]

File: src/pageObjects/formSignUp/components/SignUpPopup.js (hunk 1 above)
Comment (VasilyevDmitry) [reply src/pageObjects/formSignUp/components/SignUpPopup.js]: fixed

File: src/pageObjects/formSignUp/components/SignUpPopup.js (hunk 2 above)
Comment (VasilyevDmitry) [reply src/pageObjects/formSignUp/components/SignUpPopup.js]: fixed

File: src/pageObjects/formSignUp/FormSignUp.js (hunk 3 above)
Comment (VasilyevDmitry) [reply src/pageObjects/formSignUp/FormSignUp.js]: fixed

File: src/pageObjects/BasePage.js (hunk 4 above)
Comment (VasilyevDmitry) [reply src/pageObjects/BasePage.js]: fixed

**Review by sttaran — APPROVED** [review]

**Pull request merged**